File: library/src/main/java/com/harvey/arcface/moodel/FaceFindCameraModel.java
package com.harvey.arcface.moodel;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

//一个Camera摄像头包含的所有人脸及图片信息

public class FaceFindCameraModel implements Serializable {
	List<FaceFindModel> faceFindModels;
	byte[] cameraData;

	public FaceFindCameraModel(FaceFindCameraModel model) {
        this.cameraData = model.cameraData.clone();
        this.faceFindModels=new ArrayList<>();
        for (FaceFindModel findModel:model.faceFindModels){
			faceFindModels.add(findModel.clone());
		}

    }
	public FaceFindCameraModel(List<FaceFindModel> faceFindModels, byte[] cameraData) {
		this.faceFindModels = faceFindModels;
		this.cameraData = cameraData;
	}

	public List<FaceFindModel> getFaceFindModels() {
		return faceFindModels;
	}

	public void setFaceFindModels(List<FaceFindModel> faceFindModels) {
		this.faceFindModels = faceFindModels;
	}

	public byte[] getCameraData() {
		return cameraData;
	}

	public void setCameraData(byte[] cameraData) {
		this.cameraData = cameraData;
	}

	public FaceFindCameraModel clone() {
		return new FaceFindCameraModel(this);
	}

}

File: kotlin-library/src/main/java/com/harvey/arcface/model/FaceFindCameraModel.kt
package com.harvey.arcface.model

import android.graphics.Rect
import java.util.*

/**
 * Created by hanhui on 2018/6/1 0001 13:55
 */
data class FaceFindCameraModel(var faceFindModels: List<FaceFindModel>,
                               var cameraData: ByteArray) {

    fun clone(): FaceFindCameraModel {
        for (model in faceFindModels) model.faceRect = Rect(model.faceRect)
        return copy(faceFindModels = faceFindModels, cameraData = cameraData.copyOf())
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is FaceFindCameraModel) return false

        if (faceFindModels != other.faceFindModels) return false
        if (!Arrays.equals(cameraData, other.cameraData)) return false

        return true
    }

    override fun hashCode(): Int {
        var result = faceFindModels.hashCode()
        result = 31 * result + Arrays.hashCode(cameraData)
        return result
    }

}